sdk/core/Azure.Core/src/Shared/ModelSerializerHelper.cs

```diff
@@ -0,0 +1,23 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Runtime.CompilerServices;
+using Azure.Core.Serialization;
+
+namespace Azure.Core
+{
+    internal static class ModelSerializerHelper
```

Comment:
Why introduce a shared source file for this?  Could it be part of a public type, or its own public type?

Reply:
Of course it can, it can also just be duplicated all over the place do you have a suggestion of what you prefer?

Reply:
One reason not to put it on a public type is the logic inside of it only applies to us, I don't see customers ever using it for their models.

Reply:
Would it make sense to move this into Argument.cs, e.g. as `Argument.AssertValidFormat()`?  At some point, we will make that type public for any client author to use to access standardized exception methods across the SDK - this seems to fall into that category.

I guess it doesn't really, since it is not validating an argument.  I see we have other shared source helper types that create exceptions for us - I'll flag this as one to handle as part of the "make shared source files public" effort and it doesn't have to be addressed as part of this PR (unless you see something obvious).

Reply:
There is also `FormatException` to consider in the mix.

Reply:
To me that means something you gave me was in the wrong format meaning I expected `HH:MM:SS` and you gave me `1:5:23`.  This exception comes later like if you ask me to deserialize using `Json` format and my model passes the NotSupported check but then we try to parse the payload as json and it was really xml you will get the appropriate JsonException (specialized FormatException).

Reply:
Opened this tracking issue in case we punt on this for the v1 release https://github.com/Azure/azure-sdk-for-net/issues/38131

Reply:
I checked with @KrzysztofCwalina on this, and he advises against NotSupportedException in this case, because "If you say you want to parse JSON but pass XML, the problem is not that it happens to be no supported on this platform, version, mode, etc."

He suggests letting the parser exception propagate, e.g. STJ.JsonReaderException(), for clarity.

Reply:
that works great for deserialization but not serialization

Reply:
it also doesn't work if the format is completely unknown as we take in an ubounded string

Reply:
addressed in this pr, we are changing to FormatException https://github.com/Azure/azure-sdk-for-net/pull/38147